libsrc/leddevice/LedDeviceFactory.cpp

```diff
@@ -196,7 +196,11 @@ LedDevice * LedDeviceFactory::construct(const Json::Value & deviceConfig)
 		const std::string username = deviceConfig.get("username", "newdeveloper").asString();
 		const bool switchOffOnBlack = deviceConfig.get("switchOffOnBlack", true).asBool();
 		const int transitiontime = deviceConfig.get("transitiontime", 1).asInt();
-		device = new LedDevicePhilipsHue(output, username, switchOffOnBlack, transitiontime);
+		std::vector<unsigned int> lightIds;
+		for (size_t i = 0; i < deviceConfig["lightIds"].size(); i++) {
```

Comment:
I get a compile error on this line
hyperion-git/libsrc/leddevice/LedDeviceFactory.cpp:201:47: error: use of overloaded operator '[]' is ambiguous
      (with operand types 'const Json::Value' and 'size_t' (aka 'unsigned long'))
                        lightIds.push_back(deviceConfig["lightIds"][i].asInt());
                                           ~~~~~~~~~~~~~~~~~~~~~~~~^~
hyperion-git/dependencies/include/json/value.h:328:20: note: candidate function
      const Value &operator[]( ArrayIndex index ) const;
                   ^
hyperion-git/dependencies/include/json/value.h:333:20: note: candidate function
      const Value &operator[]( int index ) const;